src/components/Settings/OverrideRule/OverrideRuleTile.tsx

```diff
@@ -73,18 +73,23 @@ const OverrideRuleTile = ({
           })
       );
       setKeywords(keywords);
-      const users = await Promise.all(
-        rules
-          .map((rule) => rule.users?.split(','))
-          .flat()
-          .filter((userId) => userId)
-          .map(async (userId) => {
-            const res = await fetch(`/api/v1/user/${userId}`);
-            if (!res.ok) throw new Error();
-            const user: User = await res.json();
-            return user;
-          })
+      const distinctUsers = new Array(
+        ...new Set(
+          rules
+            .map((rule) => rule.users)
+            .filter((users) => users)
+            .flat()
+            .flatMap((userIds) => userIds?.split(','))
+        )
+      ).join(',');
+      const res = await fetch(
+        '/api/v1/user' +
+          (distinctUsers
+            ? `?includeIds=${encodeURIComponent(distinctUsers)}`
+            : '')
```

Comment:
You can do a `if (rule.users)` before the `const res = ...` instead of this ternary. It would be more clear and avoid an unnecessary api call if the override rule has no rule with an user.